controller/scheduler-v2/scheduler.go

```diff
@@ -446,11 +453,9 @@ func (s *Scheduler) HandleFormationChange(ef *ct.ExpandedFormation) {
 	}()
 
 	log := logger.New("fn", "HandleFormationChange")
-	if ef.App != nil {
-		log = log.New("app.id", ef.App.ID)
-	}
-	if ef.Release != nil {
-		log = log.New("release.id", ef.Release.ID)
+	if ef == nil || ef.App == nil || ef.Release == nil {
+		log.Error("invalid formation")
+		return
 	}
```

Comment:
I still feel like having it is better than not having it, so it can at least appear in our logs instead of panicking.